libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,140 @@
+import { NgControl, NgForm } from '@angular/forms';
+import { Component, ChangeDetectorRef, Self, Optional, Input } from '@angular/core';
+import { stateType, compareObjects } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput {
+    /** Sets text of control label. */
```

Comment:
How are you going to handle complex objects just image you will have:

Object payment terms

terms: PaymentTerms = [
   {id: 1, name: "2 weeks", description: "xxxx"},
   {id: 2, name: "3 weeks", description: "xxxx"},
   {id: 2, name: "4 weeks", description: "xxxx"}
]

` <fdp-checkbox [list]=terms....`

I guess for primitive string it already works. 


and when users check, I want to receive back to the application all the terms that you selected

Reply:
implementation is changed now. now only checked value is expected from user. this case will be handled in checkbox group, but checkbox will be able to set these properties now.